minio/api.py

```diff
@@ -1858,23 +1863,15 @@ def _url_open(self, method, bucket_name=None, object_name=None,
             self._delete_bucket_region(bucket_name)
 
             # Populate response_error with error response.
-            response_error = ResponseError(response)
+            response_error = ResponseError(response, method).get_exception()
 
             # In case we did not preload_content, we need to release
             # the connection:
             if not preload_content:
                 response.release_conn()
 
-            if method == 'HEAD':
-                raise response_error.head(bucket_name, object_name)
-            elif method == 'GET':
-                raise response_error.get(bucket_name, object_name)
-            elif method == 'POST':
-                raise response_error.post(bucket_name, object_name)
-            elif method == 'PUT':
-                raise response_error.put(bucket_name, object_name)
-            elif method == 'DELETE':
-                raise response_error.delete(bucket_name, object_name)
+            if method in self.supported_methods:
```

Comment:
So the reason I did this is because when I was looking at the code 
```
            if method == 'HEAD':
                raise response_error.head(bucket_name, object_name)
            elif method == 'GET':
                raise response_error.get(bucket_name, object_name)
            elif method == 'POST':
                raise response_error.post(bucket_name, object_name)
            elif method == 'PUT':
                raise response_error.put(bucket_name, object_name)
            elif method == 'DELETE':
                raise response_error.delete(bucket_name, object_name)
            else:
                raise ValueError('Unsupported method returned'
                                 ' error: {0}'.format(response.status))
```
And i thought to myself why? Honestly there's a lot of questionable things in the code and I wish I could do a whole rewrite of minio-py.